central/networkgraph/flow/datastore/internal/store/postgres/store.go

```diff
@@ -160,9 +163,10 @@ func (s *flowStoreImpl) insertIntoNetworkflow(ctx context.Context, tx *postgres.
 		obj.GetProps().GetL4Protocol(),
 		protocompat.NilOrTime(obj.GetLastSeenTimestamp()),
 		clusterID,
+		protocompat.NilOrTime(protoconv.ConvertMicroTSToProtobufTS(lastUpdateTS)),
```

Comment:
I think that `protocompat.NilOrTime` can return `nil`. Maybe we should use `protocompat.NilOrNow` - that should return time.

Also, I'm not sure if we can ever get `lastUpdateTS < 0` and what will happen in that case? Because `timestamp.MicroTS` - is `int`.

**EDIT:** The same applies for other places where `protocompat.NilOrTime` is used.

Reply:
Done. I also change the if that checks for `lastUpdateTS` == 0 to `lastUpdateTS <= 0` so the changes of getting here with an invalid timestamp are pretty much impossible.